src/snapred/backend/recipe/algorithm/PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -58,11 +66,11 @@ def chopIngredients(self, ingredients: PixelGroupingIngredients) -> None:
         self.delTheta = ingredients.instrumentState.delTh
         return
 
-    def unbagGroceries(self, ingredients: PixelGroupingIngredients):
+    def unbagGroceries(self, ingredients: PixelGroupingIngredients):  # noqa: ARG002
```

Comment:
Do we need to pass in `PixelGroupingIngredients` if they aren't being used?

Reply:
++

Reply:
My response here would be the same as previous.  An ABC for `PythonAlgorithm` would have `unbagGroceries` with an `ingredients` argument (, although I suppose it could be optional, I personally wouldn't lean towards designing it that way (, as it requires "knowing too much" in advance about the design)).

Reply:
This is probably the fault of some inconsistent design, most `unbagGroceries` methods in algos actually only have the self argument, the main purpose of the method is to assign workspaces to class members.
`chopIngredients` would take the ingredients and load them into class members.

There is the case of `PixelGroupingParametersCalculationAlgorithm` breaking this pattern, and it should really be refactored to follow suit, pushing the `self.loadNeededLogs(self.groupingWorkspaceName, ingredients)` call out of `unbagGroceries` into `PyExec` after both `chopIngredients` and `unbagGroceries` have executed.

Reply:
I do lean in favor of ABC impl, and having an explicit pass is informative even if it doesnt execute anything.

Reply:
It is inconsistent, I take the blame for it.  This method  calls the to load the logs, which needs the ingredients.  If that's no longer happening, then ingredients can be removed from this method's arguments.